File: Assets/Scripts/PlatformMove.js
#pragma strict

private var Xpos : float;
private var Ypos : float;
private var max : boolean;

var Vert : boolean;
var maxAmount : int;
var step : float;

function Start () {
	Xpos = transform.position.x;
	Ypos = transform.position.y;
}

function FixedUpdate () {

	//SET THE MAX
	if(Vert){ //Vertical
		if(transform.position.y >= Ypos + maxAmount){
			max = true;
		} else if (transform.position.y <= Ypos){
			max = false;
		}
	} else { //Horizontal
		if(transform.position.x >= Xpos + maxAmount){
			max = true;
		} else if (transform.position.x <= Xpos){
			max = false;
		}
	}

	//MOVING THE PLATFORM
	if(Vert){//Vertical movement
		if(!max){
			transform.position.y += step;
		}else {
			transform.position.y -= step;
		}
	} else {
		if(!max){
			transform.position.x += step;
		}else {
			transform.position.x -= step;
		}
	}
	
}

File: Assets/Scripts/PlatformMove.cs
using UnityEngine;
using System.Collections;

public class PlatformMove : MonoBehaviour {

	public bool goVert = false;
	public int maxAmount = 4;
	public float step = 0.05f;

	private float xPos = 0;
	private float currentX = 0;
	private float yPos = 0;
	private float currentY = 0;
	private bool atMaxPosition = false;

	void Start () {
		xPos = transform.position.x;
		yPos = transform.position.y;
		currentX = transform.position.x;
		currentY = transform.position.y;
	}
	
	void FixedUpdate () {
		//SET THE MAX
		if(goVert){ //Vertical
			if(currentY >= yPos + maxAmount){
				atMaxPosition = true;
			} else if (currentY <= yPos){
				atMaxPosition = false;
			}
		} else { //Horizontal
			if(currentX >= xPos + maxAmount){

				atMaxPosition = true;
			} else if (currentX <= xPos){
				atMaxPosition = false;
			}
		}
		
		//MOVING THE PLATFORM
		if(goVert){//Vertical movement
			if(!atMaxPosition){
				currentY += step;
				transform.position = new Vector3(currentX, currentY); 
			}else {
				currentY -= step;
				transform.position = new Vector3(currentX, currentY); 
			}
		} else {

			if(!atMaxPosition){
				currentX += step;
				transform.position = new Vector3(currentX, currentY); 
			}else {
				currentX -= step;
				transform.position = new Vector3(currentX, currentY); 
			}
		}
		
	}
}